packages/app/src/systems/DApp/hooks/useConnectRequest.tsx

```diff
@@ -1,90 +1,80 @@
-import { useInterpret, useSelector } from '@xstate/react';
+import { useSelector } from '@xstate/react';
 import { useEffect, useMemo } from 'react';
 
-import type { ConnectMachineState } from '../machines';
-import { connectMachine } from '../machines';
-import { useConnectRequestMethods } from '../methods';
+import type { ConnectRequestState } from '../machines';
 
+import { Services, store } from '~/store';
 import { useAccounts } from '~/systems/Account';
 
 const selectors = {
-  isSelectingAccounts: (state: ConnectMachineState) => {
+  isSelectingAccounts: (state: ConnectRequestState) => {
     return state.matches('connecting.selectingAccounts');
   },
-  isConnecting: (state: ConnectMachineState) => {
+  isConnecting: (state: ConnectRequestState) => {
     return state.matches('connecting.authorizing');
   },
-  origin: (state: ConnectMachineState) => {
+  origin: (state: ConnectRequestState) => {
     return state.context.origin;
   },
-  selectedAddresses: (state: ConnectMachineState) => {
+  selectedAddresses: (state: ConnectRequestState) => {
     return state.context.selectedAddresses;
   },
 };
 
 export function useConnectRequest() {
   const { account, accounts, isLoading } = useAccounts();
-  const connectionService = useInterpret(connectMachine);
-  const isConnecting = useSelector(connectionService, selectors.isConnecting);
+  const service = store.useService(Services.connectRequest);
+  const isConnecting = useSelector(service, selectors.isConnecting);
```

Comment:
No need for that as I'm using the store.useService, the result is the same.

Reply:
Ok, the reason of this comment:
1- we should have a pattern of consuming state from global machines... right now some places use `store.useSelector(Services.connectRequest, selectors.error);` and other use only `useSelector` (from xstate)
2- could there be any difference between this 2 methods? as one of them is implemented by xstate, and the other by our store...

this can led to unexpected data problems that can drain our time debugging until we figure out it's because of different ways of getting data from global state.

you think using on this 2 different ways can bring us problems @pedronauck ??

Reply:
Looking at the code of useService you notice it does the same thing. The main difference is that useSelector has both in one action, in places we use a lot of useSelectors I think the best option is todo the useService as it reduces the number of hooks also.